sqlmesh/core/model/definition.py

```diff
@@ -1028,8 +1027,10 @@ def update_schema(
             schema, default_schema=default_schema, default_catalog=default_catalog
         )
 
-        mocked_star = normalize_identifiers(SQLMESH_MOCKED_STAR, dialect=self.dialect)
-        if mocked_star.name in (self.columns_to_types or {}):
+        query = self._query_renderer.render(optimize=False)
+        if isinstance(query, exp.Expression) and any(
+            select.output_name.upper() == SQLMESH_MOCKED_STAR for select in query.selects
```

Comment:
SQLMESH_MOCKED_STAR in query.named_selects

Reply:
Way better, thanks

Reply:
Ah haha this is actually problematic because `named_selects` may have lowercase or uppercase names depending on the dialect and won't match the `SQLMESH_MOCKED_STAR`. :)

Reply:
what if we just made the mock something that's not a text, like __

Reply:
Hm not sure, it feels like a hack and looks a bit weird. It may also make it harder to search for in the codebase later if needed.

We _could_ though use the `named_selects` you suggested as:
```diff
-        if isinstance(query, exp.Expression) and any(
-            select.output_name.upper() == SQLMESH_MOCKED_STAR for select in query.selects
+        if isinstance(query, exp.Subqueryable) and any(
+            name.upper() == SQLMESH_MOCKED_STAR for name in query.named_selects
```

Reply:
sounds good